trieste/bayesian_optimizer.py

```diff
@@ -198,24 +253,16 @@ def optimize(
 
             except Exception as error:
                 tf.print(
```

Comment:
Unrelated to this PR, but I was surprised by this combination of tf.print() (whose only purpose, really, is for running inside tf.function()) and Exception handling (which I _believe_ would not play well together with tf.function() - can you confirm that?)... Is this code run both using and not using tf.function wrapper?

Reply:
very good spot. hmmm ... i don't expect we can do without exception handling, so perhaps `tf.print` is useless

Reply:
added note to https://github.com/secondmind-labs/trieste/issues/90#issuecomment-763660627